### PR TITLE
drivers:implement HiSilicon Security Engine(SEC) module.

### PR DESCRIPTION
**Pull request by LeiSen62**
HiSilicon SEC is used in security applications such as authentication and data encryption and decryption. This module implement the hardware initialization of the SEC.

### CODE REVIEW

**Comment by LeiSen62** [comment]
I`m sorry to close the previous PR by mistake. I submitted a new PR based on HPRE.Please review it again. Thank you very much.#6689

**Comment by LeiSen62** [comment]
The review comments have been modified. Please review it again. Thank you very much@etienne-lms @jenswi-linaro

File: core/drivers/crypto/hisilicon/sec_main.c [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+ * Copyright 2022-2024 HiSilicon Limited.
+ * Kunpeng hardware accelerator SEC module init.
+ */
+#include <io.h>
+#include <malloc.h>
+#include <sys/queue.h>
+#include <trace.h>
+#include <util.h>
+#include <initcall.h>
+#include "sec_main.h"
```
Comment (jenswi-linaro) [inline core/drivers/crypto/hisilicon/sec_main.c:12]: `sec_main.h` is in the include path so `<>` should be used instead.

File: core/drivers/crypto/hisilicon/include/sec_main.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/* Copyright (c) 2022-2024 HiSilicon Limited. */
+#ifndef __SEC_MAIN_H
+#define __SEC_MAIN_H
+
+#include "hisi_qm.h"
```
Comment (jenswi-linaro) [inline core/drivers/crypto/hisilicon/include/sec_main.h:6]: `hisi_qm.h` is in the include path so please use `<>` instead.

File: core/drivers/crypto/hisilicon/sec_main.c [hunk 3]
```diff
@@ -0,0 +1,273 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2022-2024 HiSilicon Limited.
+ * Kunpeng hardware accelerator SEC module init.
+ */
+#include <io.h>
```
Comment (jenswi-linaro) [inline core/drivers/crypto/hisilicon/sec_main.c:6]: Please keep the list of includes sorted.

File: core/drivers/crypto/hisilicon/sec_main.c [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+	return HISI_QM_DRVCRYPT_NO_ERR;
+}
+
+static enum hisi_drv_status sec_dev_status_check(struct hisi_qm *qm)
+{
+	uint32_t val = 0;
+
+	val = io_read32(qm->io_base + SEC_CORE_INT_SOURCE);
+	if (val & SEC_RAS_NFE_ENB_MASK) {
+		EMSG("SEC NFE RAS happened, need to reset");
```
Comment (jenswi-linaro) [inline core/drivers/crypto/hisilicon/sec_main.c:162]: Is someone supposed to read this and do something?

File: core/drivers/crypto/hisilicon/sec_main.c [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+	if (!qm->io_base) {
+		EMSG("Fail to get qm io_base");
+		return HISI_QM_DRVCRYPT_EFAULT;
+	}
+
+	qm->fun_type = sec_dev->fun_type;
+	qm->vfs_num = sec_dev->vfs_num;
+	qm->sqe_size = SEC_SQE_SIZE;
+	qm->sqe_log2_size = SEC_SQE_LOG2_SIZE;
+	if (qm->fun_type == HISI_QM_HW_PF) {
```
Comment (jenswi-linaro) [inline core/drivers/crypto/hisilicon/sec_main.c:198]: This is always true, but I assume that might change later. If this is false, does it still make sense to print something about the version or the state of the initialization?

**Comment by LeiSen62** [comment]
Hi, does this patch still need to be modified? If not, please merge it.@ jenswi-linaro

**Review by etienne-lms — COMMENTED** [review]
There are 2 comments from @jenswi-linaro that have no answers.
I've added a minor conment.

File: core/drivers/crypto/hisilicon/sec_main.c (hunk 4 above)
Comment (jenswi-linaro) [reply core/drivers/crypto/hisilicon/sec_main.c:162]: Please use DMSG() for debug info.

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`
With or without my last comment addressed.

**Comment by LeiSen62** [comment]
Hi @jforissier ,all review comments have been modified. Can you help me merge the patch?

**Comment by jforissier** [comment]
@LeiSen62 please add @jenswi-linaro's Acked-by. Perhaps @etienne-lms want s to give one, too.

**Comment by LeiSen62** [comment]
Hi @jforissier , I had added jenswi-linaro's  and etienne-lms`s Acked-by.

File: core/drivers/crypto/hisilicon/sec_main.c (hunk 4 above)
Comment (etienne-lms) [reply core/drivers/crypto/hisilicon/sec_main.c:162]: The function returns an error code so I think it's fair it uses here the `EMSG()` macro.

**Comment by jforissier** [comment]
> Hi @jforissier , I had added jenswi-linaro's and etienne-lms`s Acked-by.

@etienne-lms has not given an Acked-by yet unless I missed something.

File: core/drivers/crypto/hisilicon/sec_main.c (hunk 4 above)
Comment (LeiSen62) [reply core/drivers/crypto/hisilicon/sec_main.c:162]: I think so, and  I didn't make the change.

**Comment by LeiSen62** [comment]
Hi @etienne-lms, your review comments have been modified, Can you give an Acked-by?

**Review by etienne-lms — COMMENTED** [review]
all good to me.
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>`

**Comment by LeiSen62** [comment]
Hi @jforissier, can you help me to merge the patch?

**Comment by jforissier** [comment]
> Hi @jforissier, can you help me to merge the patch?

Done, thank you!

**Pull request merged**